pkg/transform/kubernetes/populators_test.go

```diff
@@ -588,6 +588,53 @@ func TestPopulateEnvFrom(t *testing.T) {
 	}
 }
 
+func TestPopulateVolumes(t *testing.T) {
+	volumeClaims := []spec.VolumeClaim{{Name: "foo"}, {Name: "bar"}, {Name: "barfoo"}}
+	volumes := []api_v1.Volume{{Name: "foo"}}
+
+	// a volumeMount is defined but that is not there in volumeClaims
+	// neither it is in pod level volumes, so this should fail
+	failingContainers := []api_v1.Container{
+		{VolumeMounts: []api_v1.VolumeMount{{Name: "baz"}}},
+	}
+
+	if _, err := populateVolumes(failingContainers, volumeClaims, volumes); err == nil {
+		t.Errorf("should have failed but passed for volumeMount that" +
+			" does not exist.")
+	}
+
+	passingContainers := []api_v1.Container{
```

Comment:
It's fine to be inconsistent where needed I think, let's be pragmatic and get things done I think. 

Also we can take some inspiration from python [pep8 guide about consistency](https://www.python.org/dev/peps/pep-0008/#a-foolish-consistency-is-the-hobgoblin-of-little-minds).